src/devices_models/devices/common/cost_functions.jl

```diff
@@ -386,10 +425,12 @@ function add_to_cost!(
     psi_container::PSIContainer,
     spec::AddCostSpec,
     cost_data::PSY.MultiStartCost,
-    component_name::String,
+    component::PSY.Component,
 )
+    component_name = PSY.get_name(component)
     resolution = model_resolution(psi_container)
     dt = Dates.value(Dates.Second(resolution)) / SECONDS_IN_HOUR
+    time_steps = model_time_steps(psi_container)
 
     if !isnothing(spec.fixed_cost) && spec.has_status_variable
```

Comment:
I think is better to loop over the time axis here and add each linear cost one at a time instead of passing time to the function.